### PR TITLE
feat(Form fields): improve accesibility of errors

### PR DESCRIPTION
**Pull request by atabel**
WEB-1958

- Added an icon to error messages (this makes the error message more accesible over inverse and for color-blind users)
- Added an aria-live region to announce all the fields with error when the form is submitted and there is more than one field with error
- Make sure all mandatory form fields are validated
- Form fields are validated onBlur. Previously, we only validated fields with a special format, like email.
- Fixed an error making VoiceOver read fields label twice

### CODE REVIEW

File: src/credit-card-expiration-field.tsx [hunk 1]
```diff
@@ -112,9 +113,6 @@ const CreditCardExpirationField = ({
     const {setFormError, jumpToNext} = useForm();
 
     const validate = (value: ExpirationDateValue, rawValue: string): string | undefined => {
-        if (!rawValue) {
-            return optional ? '' : texts.formFieldErrorIsMandatory || t(tokens.formFieldErrorIsMandatory);
-        }
```
Comment (atabel) [inline src/credit-card-expiration-field.tsx:117]: I've find this code repeated in most input fields, I've moved it inside `useFieldProps`

File: src/form-context.tsx [hunk 2]
```diff
@@ -151,15 +163,22 @@ export const useFieldProps = ({
...
+            if (!rawValues[name] && !optional) {
+                error = texts.formFieldErrorIsMandatory || t(tokens.formFieldErrorIsMandatory);
+            } else if (validate) {
+                error = validate(values[name], rawValues[name]);
+            }
+            setFormError({name, error});
             onBlur?.(e);
         },
-        inputRef: (input: HTMLInputElement | null) => register(name, {input, validator: validate}),
+        inputRef: (input: HTMLInputElement | null) => register(name, {input, validator: validate, label}),
```
Comment (atabel) [inline src/form-context.tsx:181]: I need the field label to be in the registrations map, to be able to use it when rendering the global form error: "Revisa los errores en los siguientes campos: nombre, email"

File: src/text-field-base.tsx [hunk 3]
```diff
@@ -336,6 +336,18 @@ export const TextFieldBase = React.forwardRef<any, TextFieldBaseProps>(
...
+                            style={labelStyle}
+                            error={error}
+                            forId={id}
+                            inputState={inputState}
+                            shrinkLabel={shrinkLabel}
+                            optional={!rest.required}
+                        >
+                            {label}
+                        </Label>
+                    )}
```
Comment (atabel) [inline src/text-field-base.tsx:350]: Moved the label before the input. This avoids a bug (I think it's a bug) in Safari/VoiceOver which made the label being read twice by screen reader (when focusing the input, and then as an independent element). This doesn't happen if the label appears before the input in the html.

File: src/cvv-field.tsx [hunk 4]
```diff
@@ -42,7 +42,6 @@ const TooltipContent = ({acceptedCards}: {acceptedCards: CardOptions}) => {
                             <Text2>
                                 {texts.formCreditCardCvvTooltipAmex || t(tokens.formCreditCardCvvTooltipAmex)}
                             </Text2>
-                            )
```
Comment (atabel) [inline src/cvv-field.tsx:45]: image

File: src/__screenshot_tests__/__image_snapshots__/form-select-screenshot-test-tsx-select-error-1-snap.png
Comment (yceballost) [inline src/__screenshot_tests__/__image_snapshots__/form-select-screenshot-test-tsx-select-error-1-snap.png:1]: in desktop, the text is not aligned in the same way as mobile, @aweell is this expected? 

is it possible to align the text line to the middle of the icon? (compare with mobile screenshot to see the difference)
image

**Review by yceballost — CHANGES_REQUESTED** [review]

File: src/__screenshot_tests__/__image_snapshots__/form-select-screenshot-test-tsx-select-error-1-snap.png
Comment (aweell) [reply src/__screenshot_tests__/__image_snapshots__/form-select-screenshot-test-tsx-select-error-1-snap.png:1]: Not expected, but also not correctly documented in the specs


the size of the icon could be 1lh to match the line height in mobile (16) but also in desktop (20) so even if the font-size increases appears aligned to the first line of text? this approach seems correct?

File: src/__tests__/form-test.tsx [hunk 5]
```diff
@@ -78,6 +78,29 @@ test('custom validator', async () => {
     );
 });
 
+test('when there are multiple errors on submit, the fields with error are anounced by screen reader', async () => {
```
Comment (pladaria) [inline src/__tests__/form-test.tsx:81]: what happens when multiple fields contain errors? All errors gets announced?

If so, can you add a test for this case?

**Review by pladaria — APPROVED** [review]

File: src/text-field-components.tsx [hunk 6]
```diff
@@ -94,15 +96,20 @@ export const HelperText = ({leftText, rightText, error, id}: HelperTextProps): J
     return (
         <>
             {leftText && (
-                <div className={classnames(styles.helperText, styles.leftText)}>
-                    <Text1 color={leftColor} regular as="p" id={id}>
+                <p className={classnames(styles.helperText, styles.leftHelperText)}>
+                    {error && (
+                        <Box as="span" paddingRight={4}>
+                            <IconInformationRegular size="1rem" color={leftColor} />
```
[the remark this answers is not in the document]
Comment (atabel) [reply src/text-field-components.tsx]: fixed

File: src/__screenshot_tests__/__image_snapshots__/form-select-screenshot-test-tsx-select-error-1-snap.png
Comment (atabel) [reply src/__screenshot_tests__/__image_snapshots__/form-select-screenshot-test-tsx-select-error-1-snap.png:1]: changed size to `1lh` (fallback to 1rem for old browsers not supporting `lh` unit)

**Review by aweell — APPROVED** [review]
LGTM!

**Review by marcoskolodny — APPROVED** [review]

**Review by yceballost — APPROVED** [review]

**Pull request merged**